src/test/java/com/endava/internship/collections/TestStudentList.java

```diff
@@ -0,0 +1,483 @@
+
+    private static Stream<Arguments> studentsStream() {
+        return Stream.of(
+                arguments(student1),
+                arguments(student2),
+                arguments(student3),
+                arguments(student4),
+                arguments(student5),
+                arguments(student6),
+                arguments(student7),
+                arguments(student8),
+                arguments(student9),
+                arguments(student10),
+                arguments(student11),
+                arguments(student12));
+    }
+
+    private void createThreads(int numberOfThreads, StudentList targetList, String action) {
+        for (int i = 0; i < numberOfThreads; i++) {
+            new Thread(threadProperties(targetList, action)).start();
+        }
+    }
+
+    private static Runnable threadProperties(StudentList targetList, String action) {
+        return () -> {
+            if (action.equalsIgnoreCase("remove by index")) {
+                targetList.remove(targetList.get(0));
+            }
+            if (action.equalsIgnoreCase("remove")) {
+                targetList.remove(0);
+            }
+            if (action.equalsIgnoreCase("add to index")) {
+                targetList.add(0, student0);
+            }
+            if (action.equalsIgnoreCase("add")) {
+                targetList.add(student0);
+            }
+            if (action.equalsIgnoreCase("clear")) {
+                targetList.clear();
+            }
+            if (action.equalsIgnoreCase("set to index")) {
+                targetList.set(0, student0);
+            }
+            System.out.println(targetList.size());
+        };
+    }
+
+    private static <T> List<Student> listOfStudents() {
+        return Arrays.asList(student1, student2, student3, student4, student5, student6, student7, student8, student9, student10, student11, student12);
+    }
+
+    private Object[] getArrayOfStudents() {
+        Object[] objects = {student1, student2, student3, student4, student5, student6, student7, student8, student9, student10, student11, student12};
+        return objects;
+    }
+
+    @BeforeEach
+    void setUp() {
+        students_List = new StudentList(listOfStudents());
+        emptyList = new StudentList();
+    }
+
+    @Test
+    @DisplayName("Constructor initialization with custom capacity")
+    public void constructorInitializationWithCustomCapacity() {
+        assertTrue(new StudentList(students_List.size()).addAll(listOfStudents()));
+    }
+
+    @Test
+    @DisplayName("Check throwing an ArrayIndexOutOfBoundsException while adding more elements than it's capacity to collection ")
+    public void checkThrowingAnExceptionWhileAddingMoreElementsThanItSCapacityToCollection() {
+        assertThrows(ArrayIndexOutOfBoundsException.class, () -> new StudentList(students_List.size() - 1).addAll(listOfStudents()));
+    }
+
+    @ParameterizedTest()
```

Comment:
here is the link of an alternative way of creating parametrized tests - https://www.baeldung.com/junit-params

no need to change anything, just fyi, this way can be more comfortable when you need to pass only few strings as parameters